lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -356,37 +356,51 @@ export class UserAgentApplication {
       scopes = this.filterScopes(scopes);
     }
 
-    this._loginInProgress = true;
-    
-    this.authorityInstance.ResolveEndpointsAsync()
-      .then(() => {
-        const authenticationRequest = new AuthenticationRequestParameters(this.authorityInstance, this.clientId, scopes, ResponseTypes.id_token, this._redirectUri, this._state);
-        if (extraQueryParameters) {
-          authenticationRequest.extraQueryParameters = extraQueryParameters;
-        }
-
-        var loginStartPage = this._cacheStorage.getItem(Constants.angularLoginRequest);
-        if (!loginStartPage || loginStartPage === "") {
-            loginStartPage = window.location.href;
-        }
-        else {
-            this._cacheStorage.setItem(Constants.angularLoginRequest, "")
-        }
-
-        this._cacheStorage.setItem(Constants.loginRequest, loginStartPage, this.storeAuthStateInCookie);
-        this._cacheStorage.setItem(Constants.loginError, "");
-        this._cacheStorage.setItem(Constants.stateLogin, authenticationRequest.state, this.storeAuthStateInCookie);
-        this._cacheStorage.setItem(Constants.nonceIdToken, authenticationRequest.nonce, this.storeAuthStateInCookie);
-        this._cacheStorage.setItem(Constants.msalError, "");
-        this._cacheStorage.setItem(Constants.msalErrorDescription, "");
-        const authorityKey = Constants.authority + Constants.resourceDelimeter + authenticationRequest.state;
-        if (Utils.isEmpty(this._cacheStorage.getItem(authorityKey))) {
-          this._cacheStorage.setItem(authorityKey, this.authority);
-        }
-
-        const urlNavigate = authenticationRequest.createNavigateUrl(scopes)  + Constants.prompt_select_account + Constants.response_mode_fragment;
-        this.promptUser(urlNavigate);
-      });
+      this._loginInProgress = true;
+      const adalIdToken = this._cacheStorage.getItem(Constants.adalIdToken);
+      var idTokenObject;
+      if(adalIdToken) {
+           idTokenObject = Utils.extractIdToken(adalIdToken);
+      }
+      if (idTokenObject && idTokenObject.upn ) {
+          console.log("ADAL's idToken exists. Extracting login information from ADAL's idToken ");
+          var extraQueryParam= "&" + Constants.login_hint+ "=" + idTokenObject.upn + "&" + Constants.domain_hint + "=" + Constants.organizations;
+          this.acquireTokenSilent([this.clientId], this.authority, this.getUser(), extraQueryParam)
```

Comment:
I assume acquireTokenSilent itself takes care of endpoints resolving before making request? If not, then why don't we resolve endpoints for this case? I assume endpoint resolving loads openid metadata and caches it for later usages?

Reply:
yes ATS is taking care of resolving endpoints even for this unified cache use case.